pkg/tests/ossm/smoke_test.go

```diff
@@ -63,6 +132,56 @@ func TestBookinfoInjection(t *testing.T) {
 	})
 }
 
+func validateStartUpProxyTime(t TestHelper, ns string) {
+	podTimes := ExtractProxyTimes(t, ns)
+	for _, podTime := range podTimes {
+		startupTime := podTime.LastTransitionTime.Sub(podTime.StartedAt)
+		t.Logf("Proxy startup time: %s", startupTime.String())
+		if startupTime > 10*time.Second {
+			t.Fatalf("Proxy startup time is too long: %s", startupTime.String())
+		}
+	}
+}
+
+func ExtractProxyTimes(t TestHelper, ns string) []PodTime {
```

Comment:
What's the benefit of extracting times into a struct and then checking. You could simply get the times and check them directly (I mean, merge validateStartupProxyTime and ExtractProxyTimes into a single function).

Reply:
Done, create only one func and make it cleaner